zerver/views/user_settings.py

```diff
@@ -74,6 +74,18 @@ def confirm_email_change(request: HttpRequest, confirmation_key: str) -> HttpRes
     }
     return render(request, 'confirmation/confirm_email_change.html', context=ctx)
 
+def start_email_change(request: HttpRequest, confirmation_key: str) -> HttpResponse:
+    try:
+        get_object_from_key(confirmation_key, Confirmation.EMAIL_CHANGE)
+    except ConfirmationKeyException as exception:
+        return render_confirmation_key_error(request, exception)
+
+    logout(request)  # In case user is logged in, this avoids the redirect home.
```

Comment:
I don't think we want to log out the user just for clicking the link, that might be pretty annoying to them. We should figure out how to run the re-auth flow even if they already have a logged in session.

Reply:
Didn't notice earlier that we have an `is_preview` that we could use to not redirect the user to home. Added a commit for that particular change.

(We'll squash the commits at the end.)